jasminewd2/index.d.ts

```diff
@@ -44,6 +44,13 @@ declare namespace jasmine {
     toThrowError(expected?: new (...args: any[]) => Error | Promise<new (...args: any[]) => Error>, message?: string | RegExp | Promise<string | RegExp>): Promise<void>;
   }
 
+  interface ArrayLikeMatchers<T extends ArrayLike<T>> extends Matchers<T> {
```

Comment:
The [Generic Constraint](https://www.typescriptlang.org/docs/handbook/generics.html#generic-constraints) syntax is a bit misleading here.

From my understanding, the `T` has two different meanings here:
`T extends ArrayLike<T>` expresses that a parameter named `T` must implement an interface called`ArrayLike` wich in turn takes a generic parameter *of the type of `T`*

So the first `T` declares the name of the parameter while the second usage in `ArrayLike<T>` uses the type of this parameter like so `ArrayLike<typeof T>`


The usage in the tests seem to prove that, but maybe I got some error in my understanding of how this works?

How else would I express that the generic parameter `T` must be of type `ArrayLike<T>/Array<T>`